### PR TITLE
MINOR: Document how to create source streams and tables

### PR DESCRIPTION
**Pull request by enothereska**
Originally reviewed as part of https://github.com/apache/kafka/pull/3490.

### CODE REVIEW

**Comment by enothereska** [comment]
cc @dguy

File: docs/streams/developer-guide.html [hunk 1]
```diff
@@ -450,21 +450,124 @@
...
-    <pre class="brush: java;">
-    StreamsBuilder builder = new StreamsBuilder();
-
-    KStream&lt;String, GenericRecord&gt; source1 = builder.stream("topic1", "topic2");
-    KTable&lt;String, GenericRecord&gt; source2 = builder.table("topic3", "stateStoreName");
-    GlobalKTable&lt;String, GenericRecord&gt; source2 = builder.globalTable("topic4", "globalStoreName");
-    </pre>
+    <h4><a id="streams_dsl_source" href="#streams_dsl_source">Creating Source Streams from Kafka</a></h4>
+
+    You can easily read data from Kafka topics into your application. We support the following operations.
```
Comment (dguy) [inline docs/streams/developer-guide.html]: Do we need a `<p>..</p>` here

File: docs/streams/developer-guide.html [hunk 2]
```diff
@@ -450,21 +450,124 @@
...
+                    Slightly simplified, in the case of a KStream, the local KStream instance of every application instance will be populated
+                    with data from only a <b>subset</b> of the partitions of the input topic. Collectively, i.e. across all application instances,
+                    all the partitions of the input topic will be read and processed.
+                </p>
+                <pre class="brush: java;">
+                    import org.apache.kafka.common.serialization.Serdes;
+                    import org.apache.kafka.streams.kstream.KStreamBuilder;
+                    import org.apache.kafka.streams.kstream.KStream;
+
+                    KStreamBuilder builder = new KStreamBuilder();
```
Comment (dguy) [inline docs/streams/developer-guide.html]: This should be `StreamBuilder` on trunk. Also on L472

File: docs/streams/developer-guide.html [hunk 3]
```diff
@@ -450,21 +450,124 @@
...
+                    with data from only a subset of the partitions of the input topic. Collectively, i.e. across all application instances, all
+                    the partitions of the input topic will be read and processed.
+                <p>
+                You may provide an optional name for the table (more precisely, for the internal state store that backs the table).
+                When a name is provided, the table can be queryied using <a href="#streams_interactive_queries">interactive queries</a>.
+                When a name is not provided the table will not queryable and an internal name will be provided for the state store.
+
+                <pre class="brush: java;">
+                    import org.apache.kafka.common.serialization.Serdes;
+                    import org.apache.kafka.streams.kstream.KStreamBuilder;
```
Comment (dguy) [inline docs/streams/developer-guide.html]: Here and below, this needs to be `StreamBuilder` for trunk

File: docs/streams/developer-guide.html [hunk 4]
```diff
@@ -450,21 +450,124 @@
...
+                    instance will get its own, full copy of the topic's data.
+
+                <p>
+                You may provide an optional name for the table (more precisely, for the internal state store that backs the table).
+                When a name is provided, the table can be queryied using <a href="#streams_interactive_queries">interactive queries</a>.
+                When a name is not provided the table will not queryable and an internal name will be provided for the state store.
+
+                <pre class="brush: java;">
+                    import org.apache.kafka.common.serialization.Serdes;
+                    import org.apache.kafka.streams.kstream.KStreamBuilder;
```
Comment (dguy) [inline docs/streams/developer-guide.html]: ditto

File: docs/streams/developer-guide.html (hunk 1 above)
Comment (enothereska) [reply docs/streams/developer-guide.html]: Not sure. Will put more `<p>s` to be safe. As it is it shows up fine for me.

File: docs/streams/developer-guide.html (hunk 2 above)
Comment (enothereska) [reply docs/streams/developer-guide.html]: Oh dear, good catch. Code is changing faster than docs.

**Review by dguy — COMMENTED** [review]
Sorry @enothereska, where `StreamsBuilder` is used, the package name should be: `org.apache.kafka.streams`

**Comment by enothereska** [comment]
This might be a nightmare to maintain.

**Review by dguy — APPROVED** [review]
Thanks @enothereska, LGTM

**Comment by dguy** [comment]
Merged to trunk

**Pull request closed without merging**